### DISCUSSION TITLE
Verifying the Clair Installation

### DISCUSSION THREAD

**Discussion by rammanoj** [quay/clair#1821] [Q&A] [answered]
I have been trying to install clair all day today. I couldn't even figure out what's happening wrong or how to verify if my installation went well. I followed the instructions given [here](https://quay.github.io/clair/howto/testing.html#starting-a-cluster). I have four containers running locally:
![image](https://github.com/quay/clair/assets/21082378/60f79bc1-8d6a-48d4-9dd1-9e35644dea54)
I visited http:localhost:8080/, and traefik is showing everything in green. So, I ran an image scan with Klar, but it failed. To verify if my clair is installed correctly. I visited http://localhost:6060/indexer and even the matcher's URL as well. It prompted me:
![image](https://github.com/quay/clair/assets/21082378/5d3b07a9-57bb-445d-86ec-1750f56fd803)

I also visited http://0.0.0.0:51221/ to check if the quay came up. But it replied "Service Unavailable". 

Is the clair actually running ? How do I verify if it's working? I could not find any help in the documentation regarding this issue.

Any help would be appreciated.

@crozzy / @hdonnay

**Comment by hdonnay** [comment]
I don't know what `klar` is. What does running a report with `clairctl report` say? What do the logs say?

Your screenshot shows a `401`, that indicates you have it configured with authentication.

**Comment by rammanoj** [comment]
@hdonnay Thanks for the reply. I guess you are right. Here is the screenshot of running clairctl:
![image](https://github.com/quay/clair/assets/21082378/3d5e0473-36bb-4cbe-bbe3-fd1a43204618)

However, I have just cloned and ran docker compose. Is authentication enabled by default ?

**Comment by hdonnay** [comment, marked as answer]
Yes, see [`local-dev/clair/config.yaml`](https://github.com/quay/clair/blob/main/local-dev/clair/config.yaml). You'll need to use that configuration. The help from the `clairctl` commands should explain how.

**Reply by rammanoj** [reply]
Thanks @hdonnay for reply. I tried it out. But I still get the same issue!

![image](https://github.com/quay/clair/assets/21082378/fb10e591-68e3-4b89-9afc-b7d0be2ce7be)

**Reply by rammanoj** [reply]
I would also like to know if port configurations are causing some issue. I believe the database port is being changed to 64566 (not by me manually). Would this cause some issue ?

![image](https://github.com/quay/clair/assets/21082378/99bf2e91-fcf6-47eb-9ac4-987b906d3cb4)

**Reply by hdonnay** [reply]
Do you have permissions for the repository `docker.io/library/romantic_hawking`? That's what the error clearly says.

Randomized host ports have no effect on the container network.

**Reply by rammanoj** [reply]
My bad! I was in assumption that it works for a local container. Thanks for pointing out!

I started the scan for a image. It has not outputted anything for some time now (~10 mins). Is it intended ?
![image](https://github.com/quay/clair/assets/21082378/8b755ecb-8735-4348-a20b-13667846e7d1)

I would also like to know if there are any official plugins to run the scan for local images / containers ? I found [this](https://github.com/arminc/clair-scanner) out. But it seems like it is not regularly maintained!

**Reply by hdonnay** [reply]
No, Clair requires layers be accessible via HTTP at whatever addresses are provided in the manifest. Indexing can take large amounts of time depending on the resources available.

**Reply by rammanoj** [reply]
Got it! Could you please help me with how to specify username / password in case of private registries ?

**Reply by hdonnay** [reply]
The standard docker and podman authentication files are consulted.

**Comment by rammanoj** [comment]
@hdonnay Thanks a lot for all the help. I was able to setup and perform a few scans with Clair. 

Initially, I tried with standard images like ubuntu:22.04. It worked and showed a few vulnerabilities. But when I tried with a sample image I developed (with a few vulnerabilities in it), it's just replying ok. 

Does it take sometime for database to actually get refreshed and match ? (It's been 3 hours since DB is running)
Here is the output upon running cliar:
![image](https://github.com/quay/clair/assets/21082378/eb8e35b2-120f-465a-90e4-1c4b1604d62e)

However, docker Hub has detected a few vulnerabilities:
![image](https://github.com/quay/clair/assets/21082378/c23ce6c2-2f9c-4624-be2e-e2360b1b0825)

**Reply by hdonnay** [reply]
There's no way to troubleshoot without logs and the JSON output.

**Reply by ssadeghh** [reply]
> @hdonnay Thanks a lot for all the help. I was able to setup and perform a few scans with Clair.
> 
> Initially, I tried with standard images like ubuntu:22.04. It worked and showed a few vulnerabilities. But when I tried with a sample image I developed (with a few vulnerabilities in it), it's just replying ok.
> 
> Does it take sometime for database to actually get refreshed and match ? (It's been 3 hours since DB is running) Here is the output upon running cliar: ![image](https://private-user-images.githubusercontent.com/21082378/254085886-eb8e35b2-120f-465a-90e4-1c4b1604d62e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2NTgyNTIsIm5iZiI6MTcyNDY1Nzk1MiwicGF0aCI6Ii8yMTA4MjM3OC8yNTQwODU4ODYtZWI4ZTM1YjItMTIwZi00NjVhLTkwZTQtMWM0YjE2MDRkNjJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI2VDA3MzkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5NTlkNzM1ODlkMTMwNmRhYWZlYmZlYzg4ZTA1NjE4Yzg4NGQ5Zjg1NTMyMjNhYzEzY2FiNzAyMjA2NDg2N2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UeFP0Mv9AtoUzhWQqtx2tDPkXjj3QIGWb8_YZOQ_Oac)
> 
> However, docker Hub has detected a few vulnerabilities: ![image](https://private-user-images.githubusercontent.com/21082378/254086071-c23ce6c2-2f9c-4624-be2e-e2360b1b0825.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2NTgyNTIsIm5iZiI6MTcyNDY1Nzk1MiwicGF0aCI6Ii8yMTA4MjM3OC8yNTQwODYwNzEtYzIzY2U2YzItMmY5Yy00NjI0LWJlMmUtZTIzNjBiMWIwODI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI2VDA3MzkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjMWRmZWI1MGViYjBiNmVjNjg4OWFiM2JiZTgzZmU2ZmU5MGZjZTE4NWJmMmFlYTEwYWQ3MWY2YjdlNTNmMzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3n4LUFj-1vboYJU1ptXUny4Eeq43Wdp0g487DFWwJdE)

hi, can you tell me how you handle authentication to scan other docker images and not just public images like ubuntu pls?

**Reply by hdonnay** [reply]
What have you tried? Docker/Podman configuration in the standard locations should work.

**Reply by ssadeghh** [reply]
I've tried Docker
